src/cls/rgw/cls_rgw_ops.h

```diff
@@ -1002,56 +1002,80 @@ struct cls_rgw_lc_get_next_entry_op {
 WRITE_CLASS_ENCODER(cls_rgw_lc_get_next_entry_op)
 
 struct cls_rgw_lc_get_next_entry_ret {
-  pair<string, int> entry;
+  cls_rgw_lc_entry entry;
 
   cls_rgw_lc_get_next_entry_ret() {}
 
   void encode(bufferlist& bl) const {
-    ENCODE_START(1, 1, bl);
+    ENCODE_START(2, 1, bl);
+    pair<string, int> entry_pair(entry.bucket, entry.status);
+    ::encode(entry_pair, bl); 
     ::encode(entry, bl);
     ENCODE_FINISH(bl);
   }
 
   void decode(bufferlist::iterator& bl) {
-    DECODE_START(1, bl);
-    ::decode(entry, bl);
+    DECODE_START(2, bl);
+    pair<string, int> entry_pair;
+    ::decode(entry_pair, bl);
+    entry.bucket = entry_pair.first;
+    entry.status = entry_pair.second;
+    if (struct_v >= 2) {
+      ::decode(entry, bl);
+    }
```

Comment:
I think it is still not in the right way.  For keeping backward compatible, maybe we can implement it in this way
```C++
struct cls_rgw_lc_get_next_entry_ret {
  pair<string, int> entry;
  cls_rgw_lc_entry detail_entry;

  void encode(bufferlist& bl) const {
    ENCODE_START(2, 1, bl);
    ::encode(entry, bl);
    ::encode(detail_entry, bl);
    ENCODE_FINISH(bl);
  }

  void decode(bufferlist::iterator& bl) {
    DECODE_START(2, bl);
    ::decode(entry, bl);
    if (struct_v >= 2) {
      ::decode(detail_entry, bl);
    }
    DECODE_FINISH(bl);
  }
};
WRITE_CLASS_ENCODER(cls_rgw_lc_get_next_entry_ret)
```

It is just an example. What do you think? @yehudasa

Reply:
@ZVampirEM77 here the problem is entry is only a part of detail_entry. entry is for backward compatibility. So on encoding, we need partly copy detail_entry to entry at first, and then 
```cpp
::encode(entry, bl); 
::encode(detail_entry, bl);
```
on decoding, we need decode entry first, copy entry to detail_entry, and if struct_v >= 2, decode detail_entry directly.

Reply:
@ZVampirEM77 the key point here is we replace pair<string, int> with cls_rgw_lc_entry, for backward compatibility, we need partly copy between pair<string, int> and cls_rgw_lc_entry, and encode or decode pair<string, int> first.